apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/McsClaimResponseTransformerV2.java

```diff
@@ -52,69 +105,137 @@ private static ClaimResponse transformClaim(PreAdjMcsClaim claimGroup) {
   }
 
+  private static Resource getContainedPatient(PreAdjMcsClaim claimGroup) {
+    return new Patient()
+        .setIdentifier(
+            List.of(
+                new Identifier()
+                    .setType(
+                        new CodeableConcept(
+                            new Coding(
+                                CommonCodings.MC.getSystem(),
+                                CommonCodings.MC.getCode(),
+                                CommonCodings.MC.getDisplay())))
+                    .setSystem(TransformerConstants.CODING_BBAPI_MEDICARE_BENEFICIARY_ID_UNHASHED)
+                    .setValue(claimGroup.getIdrClaimMbi())))
+        .setName(getBeneName(claimGroup))
+        .setGender(
+            claimGroup.getIdrBeneSex() == null
+                ? null
+                : GENDER_MAP.get(claimGroup.getIdrBeneSex().toLowerCase()))
+        .setId("patient");
+  }
+
+  private static List<HumanName> getBeneName(PreAdjMcsClaim claimGroup) {
+    List<HumanName> names;
+
+    if (claimGroup.getIdrBeneLast_1_6() != null
+        || claimGroup.getIdrBeneFirstInit() != null
+        || claimGroup.getIdrBeneMidInit() != null) {
+      HumanName name =
+          new HumanName()
+              .setFamily(
+                  claimGroup.getIdrBeneLast_1_6() == null
+                          || claimGroup.getIdrBeneLast_1_6().isEmpty()
+                      ? null
+                      : claimGroup.getIdrBeneLast_1_6().charAt(0) + ".");
+
+      if (claimGroup.getIdrBeneFirstInit() != null || claimGroup.getIdrBeneMidInit() != null) {
+        name.setGiven(
+            List.of(
+                new StringType(
+                    claimGroup.getIdrBeneFirstInit() == null
+                        ? null
+                        : claimGroup.getIdrBeneFirstInit() + "."),
+                new StringType(
+                    claimGroup.getIdrBeneMidInit() == null
+                        ? null
+                        : claimGroup.getIdrBeneMidInit() + ".")));
+      }
+
+      names = List.of(name);
+    } else {
+      names = null;
+    }
+
+    return names;
+  }
+
   private static List<Extension> getExtension(PreAdjMcsClaim claimGroup) {
-    return Collections.singletonList(
-        new Extension()
-            .setUrl("https://dcgeo.cms.gov/resources/variables/mcs-status")
-            .setValue(
-                new Coding(
-                    "https://dcgeo.cms.gov/resources/variables/mcs-status",
-                    claimGroup.getIdrStatusCode(),
-                    claimGroup.getIdrStatusCode().toLowerCase())));
+    List<Extension> extensions = new ArrayList<>();
+
+    if (claimGroup.getIdrStatusCode() != null) {
+      extensions.add(
+          new Extension(BBCodingSystems.MCS.STATUS_CODE)
+              .setValue(
+                  new Coding(
+                      BBCodingSystems.MCS.STATUS_CODE, claimGroup.getIdrStatusCode(), null)));
+    }
+
+    if (claimGroup.getIdrClaimReceiptDate() != null) {
```

Comment:
both the receipt date and status date will need new coding systems

https://bluebutton.cms.gov/resources/variables/mcs/claim-receipt-date
https://bluebutton.cms.gov/resources/variables/mcs/status-date